lang/src/org/partiql/lang/eval/visitors/VisitorTransformBase.kt

```diff
@@ -1,11 +1,18 @@
 package org.partiql.lang.eval.visitors
 
 import org.partiql.lang.domains.PartiqlAst
+import org.partiql.lang.util.checkThreadInterrupted
 
 /**
  * Base-class for visitor transforms that provides additional functions outside of [PartiqlAst.VisitorTransform].
  */
 abstract class VisitorTransformBase : PartiqlAst.VisitorTransform() {
```

Comment:
This doc comment is incredibly oblique.  Maybe something like the following:

```suggestion
/**
 * Base-class for visitor transforms that should be used by implementors.
 * [PartiqlAst.VisitorTransform]. should **not** be used directly.
 */
abstract class VisitorTransformBase : PartiqlAst.VisitorTransform() {
```

Should implementors _ever_ implement `PartiqlAst.VisitorTransform`?  If not, we should add some comment to that effect there too.

Better yet, do we need the separation or is it because the AST visitor is generated?

Reply:
https://github.com/partiql/partiql-ir-generator/issues/72

Reply:
> Should implementors ever implement PartiqlAst.VisitorTransform? If not, we should add some comment to that effect there too.

Adding guidance to explain that customers should implement `VisitorTransformBase` instead of `PartiqlAst.VisitorTransform`.

> Better yet, do we need the separation or is it because the AST visitor is generated?

The separation is needed because the AST visitor is generated.  PIG would need to be updated to entirely avoid that.